core/src/main/java/org/ehcache/Ehcache.java

```diff
@@ -413,24 +413,19 @@ public void clear() {
         }
 
         if (cacheLoader != null) {
-          Map<K, V> loaded = Collections.emptyMap();
           try {
-            loaded = (Map<K, V>) cacheLoader.loadAll(computeResult.keySet());
+            Map<? super K, ? extends V> loaded = cacheLoader.loadAll(computeResult.keySet());
+
+            for (K key : computeResult.keySet()) {
+              V value = loaded.get(key);
+              successes.put(key, value);
+              computeResult.put(key, value);
+            }
           } catch (Exception e) {
```

Comment:
The scope the try {} block seems a little off reading through this again. You didn't introduce this in this PR so it could be captured in a new issue. 

I think the intention for catching Exception to is to cover the cacheLoader.loadAll() call. That seems okay, but the try block also covers the for loop that populates the computeResult. I suppose it isn't too likely that an exception would be thrown there (lines 419-423 above) but if it did then you could up with keys in both successes and failures I think, which seems wrong.